src/property.ts

```diff
@@ -443,7 +443,7 @@ export class Property implements PropertyPath {
 
 					let conditionType: ConditionType;
 					if (errorOptions.code)
-						conditionType = ConditionType.get(errorOptions.code) || new ErrorConditionType(errorOptions.code, "error");
+						conditionType = ConditionType.get(errorOptions.code) || new ErrorConditionType(errorOptions.code, "error", null);
```

Comment:
So b/c we're creating the condition type before constructing the rule, it won't have a source. I think the rule class will get or create its condition type, so I'm thinking we may be able to simplify this to `conditionType = errorOptions.code`?

Reply:
Well, I'm not sure the logic is quite the same actually. See `ConditionRule` constructor:
```
		this.conditionType = options.conditionType ? (typeof options.conditionType === "string" ? 
			ConditionType.get(options.conditionType) : 
			options.conditionType) :
			Rule$ensureConditionType(options.name, rootType, options.category || "Error")
```

Reply:
so this should be changed to `conditionType = errorOptions.code` and then it will use the code when constructing the condition type later on (in the block of code you posted)?

Reply:
Sorry for taking so long to get back to this, but I think our most recent change of plans negates this, since we'll be storing the source on the condition rather than the condition type.

Reply:
I believe the added `null` should be removed. I think it was for the new `source` argument, and now is falling back to mapping to the optional `sets` argument.

Reply:
Fixed!